src/components/transactions/Switch/SwitchModalContent.tsx

```diff
@@ -0,0 +1,215 @@
+import { GENERAL } from 'src/utils/mixPanelEvents';
+
+import { AssetInput } from '../AssetInput';
+import { TxModalDetails } from '../FlowCommons/TxModalDetails';
+import { TxModalTitle } from '../FlowCommons/TxModalTitle';
+import { ChangeNetworkWarning } from '../Warnings/ChangeNetworkWarning';
+import { ParaswapErrorDisplay } from '../Warnings/ParaswapErrorDisplay';
+import { SupportedNetworkWithChainId } from './common';
+import { NetworkSelector } from './NetworkSelector';
+import { SwitchActions } from './SwitchActions';
+import { SwitchErrors } from './SwitchErrors';
+import { ReserveWithBalance } from './SwitchModal';
+import { SwitchRates } from './SwitchRates';
+import { SwitchSlippageSelector } from './SwitchSlippageSelector';
+import { SwitchTxSuccessView } from './SwitchTxSuccessView';
+
+interface SwitchModalContentProps {
+  selectedChainId: number;
+  setSelectedChainId: (value: number) => void;
+  supportedNetworks: SupportedNetworkWithChainId[];
+  reserves: ReserveWithBalance[];
+  selectedNetworkConfig: NetworkConfig;
+}
+
+export const SwitchModalContent = ({
+  supportedNetworks,
+  selectedChainId,
+  setSelectedChainId,
+  reserves,
+  selectedNetworkConfig,
+}: SwitchModalContentProps) => {
+  const [slippage, setSlippage] = useState('0.001');
+  const [inputAmount, setInputAmount] = useState('0');
+  const { mainTxState: switchTxState, gasLimit, txError } = useModalContext();
+  const user = useRootStore((store) => store.account);
+  const [selectedInputReserve, setSelectedInputReserve] = useState(reserves[0]);
+  const { readOnlyModeAddress } = useWeb3Context();
+  const [selectedOutputReserve, setSelectedOutputReserve] = useState(() => {
+    const gho = reserves.find((reserve) => reserve.symbol === 'GHO');
+    if (gho) return gho;
+    return reserves[1];
+  });
+  const isWrongNetwork = useIsWrongNetwork(selectedChainId);
+
+  const handleInputChange = (value: string) => {
+    if (value === '') {
+      setInputAmount('0');
+    } else if (value === '-1' && selectedInputReserve) {
+      setInputAmount(selectedInputReserve.balance);
+    } else {
+      setInputAmount(value);
+    }
+  };
+
+  const {
+    data: sellRates,
+    isLoading: ratesLoading,
+    error: ratesError,
+  } = useParaswapSellRates({
+    chainId: selectedNetworkConfig.underlyingChainId ?? selectedChainId,
+    amount: normalizeBN(inputAmount, -1 * selectedInputReserve.decimals).toFixed(0),
+    srcToken: selectedInputReserve.underlyingAsset,
+    srcDecimals: selectedInputReserve.decimals,
+    destToken: selectedOutputReserve.underlyingAsset,
+    destDecimals: selectedOutputReserve.decimals,
+    user,
+  });
+
+  if (sellRates && switchTxState.success) {
+    return (
+      <SwitchTxSuccessView
+        txHash={switchTxState.txHash}
+        amount={inputAmount}
+        symbol={selectedInputReserve.symbol}
+        outSymbol={selectedOutputReserve.symbol}
+        outAmount={normalize(sellRates.destAmount, sellRates.destDecimals)}
```

Comment:
Would be good to also apply the slippage here, so it'll show the minimum amount received